File: Python/125-Valid-Palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        i, j = 0, len(s) - 1
        while i < j:
            while not s[i].isalnum():
                i += 1
                if i >= j:
                    return True
            while not s[j].isalnum():
                j -= 1
                if i >= j:
                    return True
            if s[i].lower() != s[j].lower():
                return False
            i, j = i + 1, j - 1
        return True

File: archives/JavaScript/125-valid-palindrome.js
let isAlnum = x =>
  (x >= 'a' && x <= 'z') || (x >= 'A' && x <= 'Z') || (x >= '0' && x <= '9')

let isPalindrome = s => {
  let i = 0
  let j = s.length - 1
  while (i < j) {
    while (i < j && !isAlnum(s[i])) i++
    while (i < j && !isAlnum(s[j])) j--
    if (s[i].toLowerCase() !== s[j].toLowerCase()) return false
    i++
    j--
  }
  return true
}
